src/storm-parsers/parser/JaniParser.cpp

```diff
@@ -1432,6 +1442,16 @@ storm::expressions::Expression JaniParser<ValueType>::parseExpression(Json const
                 STORM_LOG_THROW(false, storm::exceptions::InvalidJaniException, "Unknown operator " << opstring << " in " << scope.description << ".");
             }
         }
+        if (expressionStructure.count("constant") == 1) {
```

Comment:
Here, the expression now represents an approximation of the actual expression in the jani file. This is something to discuss a bit.

Reply:
If I understand it correctly, you suggest to have something like this for PI (and the same for `e`):
```cpp
if (constantStr == "π") {
  return storm::expressions::pi();
}
```
Did I get it right?

Reply:
Indeed!

Reply:
Hello,
I started looking into implementing this and I am not sure on what would be the preferred way to go.
My current understanding is that we need to have an additional class inherited from `BaseExpression`, that we can call `ConstantValueExpression` (or anything else), and this would provide an enum for the ConstantValue to store (i.e. PI and E), together with the same methods as for the `RationalLiteralExpression` class.

Taking this path means that some more boilerplate code needs to be written (for now I only found all the classes inherited from `ExpressionVisitor`, but maybe there is more), so I want to ask whether this is what you had in mind with your comment above, before I start implementing everything that is required.

Reply:
@tquatmann your input would be greatly appreciated here

Reply:
I personally would indeed tend to use a ConstantValueExpression or maybe a RealLiteralExpression?

Reply:
What you describe is the most correct and most complete way to implement this. However, unless someone sees an important use case for π and e outside of jani models, I suggest to reduce the mentioned boilerplate code a bit:

- Put your new expression class [here](https://github.com/moves-rwth/storm/tree/master/src/storm/storage/jani/expressions), where we collect all expressions that are somewhat special for JANI. Given the current naming scheme, the new class should have suffix `...LiteralExpression`, maybe `TranscendentalNumberLiteralExpression` ? 

- Extend the [JaniExpressionVisitor](https://github.com/moves-rwth/storm/blob/master/src/storm/storage/jani/visitor/JaniExpressionVisitor.h) and every class that inherits from this (fortunately, these are only a hand full and it's almost always clear what to do).

- Extend the `substituteConstantsInPlace` function of the JANI model, so that it can (optionally, controlled by an argument) approximate transcendentalNumberLiterals. Apply this substitution, before building the model.

The advantage of this approach is that you can still export the parsed jani without approximating the constants.
For example, when using `storm-conv --jani input.jani --constants N=4 --tojani output.jani`, the constants should still be present.

Reply:
Indeed, having the new `...LiteralExpression` class in `jani/expressions` seems more appropriate for the current PR, thanks for your suggestion!

Reply:
Hi @tquatmann ,

I just pushed a first attempt at integrating the `TranscendentalNumberLiteralExpression` in the JaniParser.
I am not sure this is 100% what you had in mind, since it resulted in an unexpected amount of changes around the code. Please let me know if there is anything that you think should be done differently.

Also, the unit tests and all the remaining comments will come at a later commit. This is just to make sure the changes related to this comment match your requests.

Reply:
I just pushed some more changes I needed to do in order to have storm-conv working as you described.
Somehow this ended up touching an unexpected amount of files, I hope this is still fine with you, but of course I am open to any suggestion you might have.

In the next days I will address the last few comments you had (namely a missing type check and accepting integers as argument of trigonometric operators).